org.jacoco.core.test/src/org/jacoco/core/instr/InstrumenterTest.java

```diff
@@ -371,6 +371,31 @@ public void testInstrumentAll_BrokenZipEntry() throws IOException {
 		}
 	}
 
+	/**
+	 * @see org.jacoco.core.analysis.AnalyzerTest#testAnalyzeAll_analyzeZip_nextEntry_IllegalArgumentException()
+	 */
+	@Test
+	public void testInstrumentAll_instrumentZip_nextEntry_IllegalArgumentException()
+			throws IOException {
+		final ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+		final ZipOutputStream zip = new ZipOutputStream(buffer);
+		zip.putNextEntry(new ZipEntry("entry"));
+		zip.closeEntry();
+		zip.close();
+		final byte[] zipBytes = buffer.toByteArray();
+		// non-UTF-8 character in entry name:
+		zipBytes[31] = (byte) 0xFF;
+
+		final ByteArrayOutputStream out = new ByteArrayOutputStream();
```

Comment:
```suggestion
		final OutputStream out = new ByteArrayOutputStream();
```

Reply:
@marchof quoting https://openjdk.org/projects/amber/guides/lvti-style-guide#G5

> Don’t worry too much about “programming to the interface” with local variables.

and we already had a discussion about a similar case in https://github.com/jacoco/jacoco/pull/1472#discussion_r1229685316 😉